src/commands/calc.rs

```diff
@@ -0,0 +1,74 @@
+use crate::commands::WholeStreamCommand;
+use crate::prelude::*;
+use nu_errors::ShellError;
+use nu_protocol::{Primitive, ReturnSuccess, Signature, SyntaxShape, UntaggedValue, Value};
+
+pub struct Calc;
+
+impl WholeStreamCommand for Calc {
```

Comment:
@coolshaurya - Something you might want to do is to make Calc a `PerItemCommand`. This would let you use $it and pass it to calc. That way, you could calculate something for each row of the table rather than reading the whole table and getting one value out.

Reply:
Currently, `calc` takes a string as an argument and processes that into a `Decimal`. So, just to clarify you want me to change the `calc` command so that it accepts a table/list of strings where the `run()` function will get each row value as an input and then it outputs a list of `Decimal`.

Reply:
When you execute per item, I believe it will just give you a series of inputs, not just one table.

Reply:
That's right. If you change to per-item, the command will get called once per row in the table.  This will let you more easily calculate the rows.

One example people might want to do is to use the `format` command to create the string and the `calc` command to calculate it.

`| format "{length} + 10" | calc $it`